File: src/test/java/org/jsoup/JsoupTest.java
package org.jsoup;

import org.jsoup.integration.ParseTest;
import org.jsoup.nodes.Document;
import org.jsoup.parser.Parser;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.file.Path;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class JsoupTest {
    // Tests for the Jsoup class. Mostly for code coverage for methods that haven't been covered elsewhere already.

    @Test
    void parseWithPath() throws IOException {
        // parse(Path path, @Nullable String charsetName, String baseUri)
        Path path = ParseTest.getPath("/htmltests/medium.html");
        Document doc = Jsoup.parse(path, "UTF-8", "https://example.com/");
        String title = "Medium HTML";
        assertEquals(title, doc.title());

        // parse(Path path)
        doc = Jsoup.parse(path);
        assertEquals(title, doc.title());

        // (Path path, @Nullable String charsetName, String baseUri, Parser parser)
        doc = Jsoup.parse(path, "UTF-8", "https://example.com/", Parser.htmlParser());
        assertEquals(title, doc.title());
    }
}

File: src/test/kotlin/org/jsoup/integration/jsoupTest.kt
package org.jsoup.integration

import org.jsoup.Jsoup
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class JsoupTest {
    @Test
    fun jsoupTest() {
        val doc = Jsoup.parse("<b>jsoup")
        doc.body()
        // It's odd to me that (prior to adding notnulll annotations), the Kotlin compiler doesn't NPE warn on this, as .parse() is !, which includes ?.
        // And compiler / Intellij doesn't have a code inspect for it either. At least have it be an option
        Assertions.assertEquals("<b>jsoup</b>", doc.body().html())
    }
}
